crates/core/src/ring_switch/eq_ind.rs

```diff
@@ -89,56 +74,44 @@ where
 		})
 	}
 
-	pub fn multilinear_extension<P: PackedField<Scalar = F>>(
+	pub fn multilinear_extension<P: PackedField<Scalar = F> + PackedExtension<FSub>>(
 		&self,
-	) -> Result<MultilinearExtension<P>, Error> {
+	) -> Result<MultilinearExtension<P>, Error>
+	where
+		F: ExtensionField<FSub>,
+	{
 		let mut evals = zeroed_vec::<P>(1 << self.z_vals.len().saturating_sub(P::LOG_WIDTH));
 		evals[0].set(0, self.mixing_coeff);
 		tensor_prod_eq_ind(0, &mut evals, &self.z_vals)?;
-		evals.par_iter_mut().for_each(|val| {
-			*val = P::from_scalars(
-				val.iter()
-					.map(|v| inner_product_subfield(v, &self.row_batch_coeffs)),
-			);
-		});
-		Ok(MultilinearExtension::new(self.z_vals.len(), evals)?)
-	}
-}
 
-#[inline(always)]
-fn inner_product_subfield<FSub, F>(value: F, row_batch_coeffs: &RowBatchCoeffs<F>) -> F
-where
-	FSub: Field,
-	F: ExtensionField<FSub>,
-{
-	if TypeId::of::<FSub>() == TypeId::of::<BinaryField1b>() && can_iterate_bytes::<F>() {
-		// Special case when we are folding with 1-bit coefficients.
-		// Use partial sums lookup table to speed up the computation.
+		let subfield_vector = <P as PackedExtension<FSub>>::cast_bases(&evals);
 
-		struct Callback<'a, F> {
-			partial_sums_lookup: &'a [F],
-			result: F,
-		}
+		let subfield_vector_mle = MLEEmbeddingAdapter::<_, P, _>::from(MultilinearExtension::new(
+			self.z_vals.len() + F::LOG_DEGREE,
+			subfield_vector,
+		)?);
 
-		impl<F: Field> ByteIteratorCallback for Callback<'_, F> {
-			#[inline(always)]
-			fn call(&mut self, iter: impl Iterator<Item = u8>) {
-				for (byte_index, byte) in iter.enumerate() {
-					self.result += self.partial_sums_lookup[(byte_index << 8) + byte as usize];
-				}
-			}
-		}
+		let extra_queries_factor: F = self.row_batch_coeffs.query[F::LOG_DEGREE..]
```

Comment:
Can you explain what's going on here? I expected just an `evaluate_partial_low` call with the `row_batch_coeffs`.